### ISSUE TITLE
Port to Linux and Windows

### ISSUE THREAD

**Issue by Nyco** [nick-denry/Chimeverse#5]
Hi,

Since Chimeverse is based on Electron, I guess (not sure) it can be quite effortless to port it to Linux and Windows.

What needs to be done?

Regards

**Comment by nick-denry** [comment]
Hi, @Nyco! 

> I guess (not sure) it can be quite effortless to port it to Linux and Windows.

You're quite right, porting to Win and/or Linux isn't big thing to do. It needs some work to improve app menu and integration. 

Well, otherwise we have perfect feautured Gajim under Win and Linux which is pretty easy to install there. At moment Chimeverse was created I've not found any macOS xmpp client with MAM and OMEMO support alltogether (except Gajim which required alot of work to run under macOS), thats why I've create it.

Since that OMEMO support was also added to PSI, but MAM support issue is still unresolved https://github.com/psi-plus/main/issues/618

So, I think Chimeverse is still could be useful for macOS and at moment I'm working at #1 and #2 with few another UI and settings improvements. But I'm not sure if Chimeverse will be useful for Win and Linux.

Do you still think it needs to be ported?

**Comment by Nyco** [comment]
> You're quite right, porting to Win and/or Linux isn't big thing to do. It needs some work to improve app menu and integration.

OK, I can help with some tests.

> Well, otherwise we have perfect feautured Gajim under Win and Linux which is pretty easy to install there. At moment Chimeverse was created I've not found any macOS xmpp client with MAM and OMEMO support alltogether (except Gajim which required alot of work to run under macOS), thats why I've create it.

Oh wow, that is a very respectful positioning! Congrats! :) Gajim indeed has always had a macOS port in progress, a few maintainers followed, but given GTK constraints it is not that easy.

I used to love both Gajim and Psi, I still use both intensively for their XML/XMPP consoles, which are very useful for testing and validation purposes.

Unfortunately, Psi and Gajim have roots in the ICQ/MSN/AIM/Yahoo!Messenger era: they are outdated in terms of interface and the purposes they serve, that is desktop-only interpersonal chat.

We have passed the era of mobile-only interpersonal chat, incarnated by WhatsApp, Telegram, FB Messenger, and millions more.

We are now in the era of team chat, or community/professional/enterprise instant messaging. The goal here is team-oriented communication and collaboration, that's why groupchats come first, are archived and searchable. The apps are on the desktop first and mobile second.

Converse fits that category of team chat (or goes in this direction), Psi and Gajim are far from team chat and would need a complete overhaul.

So Converse is not positioned against Psi and Gajim.

That's why I do believe that Chimeverse needs to exist on the desktop, and the desktop is Mac+Lin+Win, all three together. Like mobile means iOS+Android.

Chimeverse on Windows, Linux and macOS would offer choice for the user: browser or app. Like what other team chats provide. This would rock for Converse.

**Comment by nick-denry** [comment]
Well, providing a choise is always a good way to go. Even if it would be just yet another xmpp client under Win/Lin. 

Maybe it could be useful from the unified UI point-of-view. So, since cost of porting should not be too high I'll back to this issue right after #1 and #2.

**Comment by nick-denry** [comment]
> Chimeverse on Windows, Linux and macOS would offer choice for the user: browser or app. Like what other team chats provide. This would rock for Converse.

And btw yes. I've used Convese from the just-local-page and was thinking about integration of Converse to Nextcloud before create standalone app. There is a some project of that at gitlab https://gitlab.com/lil5/nextcloud-conversejs, but it's not an official.

Then I realise that in my case In fact standalone app is better than just browser cause of better system notification support and it not makes you to switch tabs all the time. But defenitely this is a power of Converse and providing a choise.

**Comment by ernierasta** [comment]
My 2 cents: multiplatform Chimeverse would be great for those, who already customized Converse and would like to have DT app "for free". So If you guys manage too make it happen, it would be great!

**Comment by nick-denry** [comment]
Hi!

A very first cross-platfom build accomplished.

Release info: https://github.com/nick-denry/Chimeverse/releases/tag/v0.1.52

I can't perform full tests for any platform atm (especially win), but every basic thing works for me at the first look. The first thing to rework is an icons now. Anyway, tests and issue reports are welcome.

**Comment by nick-denry** [comment]
Hi, @Nyco!

I guess with a new [cross-platform icons-hotfix release](https://github.com/nick-denry/Chimeverse/releases/tag/v0.1.53) this issue could be closed?

**Comment by nick-denry** [comment]
Closed due to inactivity. Feel free to reopen or comment.

**Comment by poVoq** [comment]
Could you please also make a simple tar.gz release for non-debian Linux?

**Comment by nick-denry** [comment]
> Could you please also make a simple tar.gz release for non-debian Linux?

Sure. Please create another issue so it can be attached to further release milestone.
